codec-http2/src/main/java/io/netty/handler/codec/http2/Http2CodecUtil.java

```diff
@@ -190,6 +192,148 @@ public static void writeFrameHeader(ByteBuf out, int payloadLength, byte type,
     }
 
     /**
+     * Provides the ability to associate the outcome of multiple {@link ChannelPromise}
+     * objects into a single {@link ChannelPromise} object. This can be done with relatively low overhead
+     * if the number of {@link ChannelPromise} objects is known a priori.
+     */
+    static class SimpleChannelPromiseAggregator implements ChannelFutureListener {
+        private final ChannelHandlerContext ctx;
+        private final ChannelPromise promise;
+        private final int expectedCount;
+        private int completedCount;
+
+        SimpleChannelPromiseAggregator(ChannelHandlerContext ctx, int expectedCount, ChannelPromise promise) {
```

Comment:
Rather than passing in `expectedCount` could we just internally keep track of the number of calls to `newPromise`?